server/controllers/auth/signup.js

```diff
@@ -0,0 +1,83 @@
+// modules 
+const bcrypt = require('bcrypt');
+const uuid = require('uuid/v1');
+const jwt = require('jsonwebtoken');
+
+const database = require("../../database");
+
+exports.signup = (req, res, next) => {
+  const email = req.body.email;
+  // password cannot be longer than 72 characters
+  const password = req.body.password;
+
+  // validate member existance in database
+  database.query(`
+    SELECT
+      email_address
+    FROM
+      member
+    WHERE
+      email_address = '${email}';
+  `).then(result => {
+    /**
+     * res.rows
+     * returns array of object
+     */
+    if (result.rows[0]) {
+      res.status(409).send({
+        "status": {
+          "code": 409,
+          "type": "error"
+        },
+        user: result.rows[0],
+        "error": {
+          "code": "invalid_email",
+          "message": "E-Mail already taken" 
+        }
+      })
+      next();
+    } else {
+      // password hashing
+      const bcryptSaltRounds = 10;
+      const bcryptSalt = bcrypt.genSaltSync(bcryptSaltRounds);
+      const passwordHash = bcrypt.hashSync(password, bcryptSalt);
+
+      // generate unique indentification
+      const memberId = uuid(email)
+
+      // save member to database
+      database
+        .query(`
+          INSERT INTO
+            member(member_id, email_address, password)
+          VALUES
+            ('${memberId}', '${email}', '${passwordHash}')
+          RETURNING
+            member_id, email_address, created_at;
+        `)
+        .then(result => {
```

Comment:
`await` would improve code readability.

Reply:
Can you share any small example?

I did tried to break the code into multiple lines to make it readable.

Reply:
Something like:
```js
const insertedUser = await DB.query(query);
const authToken = jwt.sign(insertedUser, 'scretKey', { expiresIn: '1h' });
```

Reply:
I am trying to keep the code consistent by using promises everywhere.

Using `async/await` at one place will make it in-consist.

Reply:
Not at one place. I suggest is to use async await everywhere. 

Let's keep going for now until first release. We could re-think these later.